internal/chain/create_key_book.go

```diff
@@ -14,77 +14,63 @@ type CreateKeyBook struct{}
 
 func (CreateKeyBook) Type() types.TxType { return types.TxCreateKeyBook }
 
-func checkCreateKeyBook(st *StateManager, tx *transactions.GenTransaction) ([]*protocol.SigSpec, *url.URL, error) {
+func (CreateKeyBook) Validate(st *StateManager, tx *transactions.GenTransaction) error {
 	if _, ok := st.Sponsor.(*state.AdiState); !ok {
-		return nil, nil, fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
+		return fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
 	}
 
 	body := new(protocol.CreateSigSpecGroup)
 	err := tx.As(body)
 	if err != nil {
-		return nil, nil, fmt.Errorf("invalid payload: %v", err)
+		return fmt.Errorf("invalid payload: %v", err)
 	}
 
 	if len(body.SigSpecs) == 0 {
-		return nil, nil, fmt.Errorf("cannot create empty sig spec group")
+		return fmt.Errorf("cannot create empty sig spec group")
```

Comment:
Should we change name to Key Book here in the error.

Reply:
Yes, but I want to do that as part of [AC-528](https://accumulate.atlassian.net/browse/AC-528)